### PR TITLE
Add a contributing guide

### LINKED ISSUE #529: Move developer instructions to a new contributing guide
### Describe your use-case which is not covered by existing documentation.

Move the instructions at the end of the README to a separate contributing guide, modeled after the guides that are referenced in the ["Improve a Plugin" tutorial](https://www.jenkins.io/doc/developer/tutorial-improve/add-a-contributing-guide/)

### Reference any relevant documentation, other materials or issues/pull requests that can be used for inspiration.

See the tutorial for references of the types of topics to include.  Some suggestions might include:

* Build and test
* Coverage reporting
* ...

### PR DESCRIPTION
**Pull request by AmanGupta677**
Moved contribution related text from the end of the README to a new file and added a contributing topic in the readme file which point to the new CONTRIBUTING file.
fixes #529 

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Contributing
+================================
```
Comment (MarkEWaite) [inline CONTRIBUTING.md:2]: With the leading `#` character, no need for the extra line of `=` characters
```suggestion
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Thanks for the contribution.  Some changes that need to be made.

**Comment by AmanGupta677** [comment]
I have made the changes that you have requested.

**Review by MarkEWaite — APPROVED** [review]
Thanks!

**Pull request merged**